src/aiida/cmdline/groups/dynamic.py

```diff
@@ -177,7 +177,7 @@ def list_options(self, entry_point: str) -> list:
             # ``typing.Union[str, None].__args__`` will return the tuple ``(str, NoneType)``. So to get the real type,
             # we simply remove all ``NoneType`` and the remaining type should be the type of the option.
             if hasattr(field_info.annotation, '__args__'):
-                args = list(filter(lambda e: e != type(None), field_info.annotation.__args__))
```

Comment:
This is one of the few changes, I think this is correct, but please check.

Reply:
The only way a different result can be achieved is when a custom type overwrites the equal function. That is not the case for `NoneType`, so it is ok.